src/main/java/com/taf/business/pages/selenium/BasePage.java

```diff
@@ -0,0 +1,38 @@
+package com.taf.business.pages.selenium;
+
+import com.taf.business.driver.DriverManager;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+    WebDriver driver;
+
+    public BasePage(WebDriver driver) {
+        this.driver = DriverManager.getDriver();
+        PageFactory.initElements(driver, this);
+    }
+
+    public void waitVisibilityOfElement(final WebElement element) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.visibilityOf(element));
+    }
+
+    public void clickOnElementUsingJs(final WebElement element) {
```

Comment:
It would be better to move JS methods to the utility class

Reply:
Thanks for suggestion, utility class was added.